src/components/LoginPage/LoginPage.js

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import ls from 'local-storage';
+
+import { Button } from '@/shared';
+import { VALIDATION_FAILED } from '../../constants/notificationData';
+
+import './LoginPage.scss';
+
+const CN = 'login-page';
+
+class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      emailToLogin: '',
+      passwordToLogin: '',
+      emailToRegister: ''
+    };
+
+    this.formLogin = React.createRef();
+    this.formRegister = React.createRef();
+    this.validate = this.validate.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      emailToLogin: ls.get('emailToLogin') || '',
+      passwordToLogin: ls.get('passwordToLogin') || '',
+      emailToRegister: ls.get('emailToRegister') || ''
+    });
+  }
+
+  saveToLocalStorage() {
+    const emailToLoginValue = this.formLogin.current[0].value;
+    const passwordToLoginValue = this.formLogin.current[1].value;
+    const emailToRegisterValue = this.formRegister.current[0].value;
+
+    ls.set('emailToLogin', emailToLoginValue);
+    ls.set('passwordToLogin', passwordToLoginValue);
+    ls.set('emailToRegister', emailToRegisterValue);
+  }
+
+  validate({ target }) {
+    const { showMessage } = this.props;
+    const clickedButton = target.innerText.toUpperCase();
+    let isInputValid = false;
+
+    if (clickedButton === 'LOGIN') {
+      isInputValid = this.formLogin.current.reportValidity();
```

Comment:
FYI you can do something like this too
```suggestion
      isInputValid = (clickedButton === 'LOGIN' ? this.formLogin : this.formRegister).current.reportValidity();
```